Adding callhome extention to installer

brucechen94539 opened the pull request:
Please review on the syntax of newly added json file to all three installers.
Will also add Shiva for reviewing with his callhome R&D expertise.

kermit-artifactory-elb-2128 (https://console.aws.amazon.com/ec2/v2/home?region=us-east-1#LoadBalancers:sort=loadBalancerName) has a 6.8.7 instance that verified with the terraform template, which contains installer-info.json on the designated location as tech spec required.

jainishshah17 commented:
@brucechen94539 remove  `userdata_secondary.sh` file as we are not using it anymore

brucechen94539 commented:
@jainishshah17, cleaned up userdata_secondary.sh from the latest checkin
@shrajfr12, added the featureId with at least one entry to ensure API could work properly

@jainishshah17, please check if the syntax for the featureID is able to pull the proper Artifactory Version from each of the installers.

shrajfr12 commented on AzureResourceManager/scripts/install_artifactory.sh:
```diff
@@ -209,6 +209,20 @@ cat <<EOF >/var/opt/jfrog/artifactory/etc/binarystore.xml
+  "features": [
+  {
+    "featureId": "Artifactory/${artifactoryVersion}",
+    "attributes": {}
```
Since attributes are optional, you can skip the empty attributes setting

brucechen94539 replied:
line removed

shrajfr12 commented on CloudFormation/artifactory-enterprise.json:
```diff
@@ -902,6 +902,21 @@
+                  "  \"features\": [\n",
+                  "  {\n",
+                  "    \"featureId\": \"Artifactory/", { "Ref": "ArtifactoryVersion" }, "\"\n",
+                  "    \"attributes\": {}\n",
```
Since attributes are optional and it is empty here, you can skip that.

brucechen94539 replied:
line removed

shrajfr12 commented on CloudFormation/artifactory-enterprise.json:
```diff
@@ -1254,6 +1269,16 @@
+                "content" : { "Fn::Join" : ["", [
+                  "{\n",
+                  "  \"productId\": \"JFrogInstaller_Cloudformation_MySQL/1.0.0\"\n",
+                  "}\n"
```
This one needs the required featureId .

brucechen94539 replied:
attribute is removed from json

shrajfr12 commented on Terraform/userdata.sh:
```diff
@@ -172,6 +172,18 @@ server {
+  "features": [
+  {
+    "featureId": "Artifactory/${artifactory_version}",
+    "attributes": {}
```
You can skip the optional empty attribute

brucechen94539 replied:
line removed

jainishshah17 commented:
@brucechen94539 Can you bump up Artifactory version in cloud formation and Terraform

brucechen94539 commented:
@jainishshah17 , bumped the version for CF and TF to 6.9.0, on Terraform/variables.tf and CloudFormation/artifactory-enterprise.json files.

shrajfr12 commented:
LGTM
